src/libsnark.ml

```diff
@@ -412,8 +421,28 @@ struct
     let to_repr t : Repr.t =
       match to_affine t with None -> Zero | Some t -> Non_zero t
 
+    let subgroup_check t =
+      match Checks.subgroup with
+      | `No_check_required ->
+          ()
```

Comment:
do you not need to check the point isn't the identity?

Reply:
See line 440. The identity is represented by a separate variant and can't be represented using affine coordinates.